### PR TITLE
feat(genesis): use embedded genesis config for default chains

### PR DESCRIPTION
**Pull request by jakim929**
- delete original genesis files
- default orchestrator config now brings in the embedded pre-generated genesis states
- instead of SourceChainID, ChainConfig now has a L2Config to signify if it is an L2 or not

## Summary by CodeRabbit

- **New Features**
  - Introduced a streamlined configuration structure for `Anvil`, removing the intermediate `Config` type.
  - Added functionality to directly retrieve the Ethereum code at a specified address with the new `EthGetCode` method.
  - Enhanced blockchain chain configuration by integrating dynamic values from the genesis deployment package.

- **Bug Fixes**
  - Simplified logic for managing genesis states and chain configurations within the `Orchestrator`.

- **Tests**
  - Added a new test to validate the deployment status of essential components in the Layer 1 environment.

- **Chores**
  - Updated dependencies to include `github.com/ethereum-optimism/go-ethereum-hdwallet`.

### CODE REVIEW

**Comment by jakim929** [comment]
* **#56** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/supersim/56?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: anvil/anvil.go [hunk 1]
```diff
@@ -272,6 +262,10 @@ func (a *Anvil) Web3ClientVersion(ctx context.Context) (string, error) {
 }
 
 // eth_ API
+func (a *Anvil) EthGetCode(ctx context.Context, account common.Address) ([]byte, error) {
```
Comment (hamdiallam) [inline anvil/anvil.go:265]: not imporant in this PR. I think externally we should just use `ethclient` externally instead of creating these stubs here

**Review by hamdiallam — APPROVED** [review]
Looks good!

File: anvil/anvil.go (hunk 1 above)
Comment (jakim929) [reply anvil/anvil.go:265]: yeah that makes sense!

**Pull request merged**